### PR TITLE
Enhance directions

### PR DESCRIPTION
**Pull request by khusbuchandra**
Added Itinerary component to display the route instruction. Saves the route and provides feature to reverse route.

## Issue Description:
Enhance directions view and capabilities:

-  Add feature to save directions 

- Add reverse directions feature 

- Include alternate routes in preview

- Customize routing text pane: partial map and text view,  and hidden

### Summary of solution:
1.  Used Leaflet easy button plugin
2.  Created an Itinerary component for displaying the instructions
3.  Saving the selected route with the backend api.

### Can this issue be closed?
No
### Should any new issues be added as a result of this solution?

1. Styling of the Itinerary Component.
2. On Mouseover of the instruction, it should display position kin map.
3. Fetching saved directions

### Have you named your branch in a descriptive way? Remember to name your branch in a unique and descriptive manner in order to properly reflect the issue or feature.
Yes, its 'enhance-directions-part1

### CODE REVIEW

**Comment by motosharpley** [comment]
I created a new branch for directions-feature that we can use to build this out further. I also adjusted the base for this pr so that it will go to the directions-feature branch when we merge.

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
         "express": "^4.16.2",
         "helmet": "^3.9.0",
         "jsonwebtoken": "^8.1.1",
+        "leaflet-easybutton": "^2.3.0",
```
Comment (jkwening) [inline package.json]: Maybe I'm missing something but I think this needs to be installed as a dependency for the client side not the server side

**Comment by jkwening** [comment]
@khusbuchandra looks goods - I fixed the module issue that I noted and merged in accordingly. Another thing to add for next steps is using different color for alternative route; something lighter than the color being used for the main directions. Thanks.

File: package.json (hunk 1 above)
Comment (khusbuchandra) [reply package.json]: Ah.. I completely missed it, Yes you are right it should be in Client. Thanks @jkwening . I think I was too lost in directions to notice it :-)

**Pull request merged**